### PR TITLE
Add undocumented methods

### PR DESCRIPTION
**Pull request by manuelmeurer**
As discussed in #21. Sorry it took a while!

I tried to keep the method, argument and error descriptions as similar to the existing ones as possible.

### CODE REVIEW

File: tasks/update.rake [hunk 1]
```diff
@@ -2,12 +2,16 @@ require_relative 'lib/slack_api/methods_spider'
 require_relative 'lib/slack_api/events_spider'
 require_relative 'lib/slack_api/spec_validator'
 
+def delete_files_except_undocumented(*dirs)
```
Comment (dblock) [inline tasks/update.rake]: As a rule of thumb you should put these in a module otherwise it pollutes global namespace. NBD

**Comment by dblock** [comment]
Maybe add to https://github.com/slack-ruby/slack-api-ref/blob/master/CHANGELOG.md and something in README about undocumented methods? Then feel free to merge.

File: tasks/update.rake (hunk 1 above)
Comment (manuelmeurer) [reply tasks/update.rake]: Yeah that was my initial idea as well but it felt like overkill for a single two-line method...

**Comment by manuelmeurer** [comment]
Will do!

File: tasks/update.rake (hunk 1 above)
Comment (dblock) [reply tasks/update.rake]: But we're better than this! I think a better way to fix this is to add an actual Rake task that deletes existing files and takes a parameter for the folder(s), then you can `invoke` it here, much cleaner than a random method.

**Comment by manuelmeurer** [comment]
@dblock How about this?

**Comment by dblock** [comment]
Very nice, merged.

**Comment by dblock** [comment]
Want to join the maintainer crew @manuelmeurer?

**Comment by manuelmeurer** [comment]
Yeah sure, sounds good! 😄

**Comment by dblock** [comment]
I'll try to use this in slack-ruby-client after https://github.com/slack-ruby/slack-ruby-client/pull/181.

**Pull request merged**